From aeeb505ad1518609dc847495872284a9bd7826aa Mon Sep 17 00:00:00 2001
From: Artur Wieczorek <artwik@wp.pl>
Date: Fri, 3 Apr 2015 21:50:38 +0200
Subject: [PATCH 4/4] Handle wxLongLong data in wxConvertAnyToVariant only if
 wxUSE_LONGLONG is enabled.

Code responsible for handling wxLongLong datatype in wxConvertAnyToVariant() should be active only if wxUSE_LONGLONG is enabled.
---
 src/common/any.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/common/any.cpp b/src/common/any.cpp
index 4165349..13f3b82 100644
--- a/src/common/any.cpp
+++ b/src/common/any.cpp
@@ -146,7 +146,7 @@ bool wxConvertAnyToVariant(const wxAny& any, wxVariant* variant)
     // and others to "longlong".
     if ( wxANY_CHECK_TYPE(any, signed int) )
     {
-#ifdef wxLongLong_t
+#if defined(wxLongLong_t) && wxUSE_LONGLONG
         wxLongLong_t ll = 0;
         if ( any.GetAs(&ll) )
         {
@@ -264,7 +264,7 @@ bool wxAnyValueTypeImplInt::ConvertValue(const wxAnyValueBuffer& src,
     wxAnyBaseIntType value = GetValue(src);
     if ( wxANY_VALUE_TYPE_CHECK_TYPE(dstType, wxString) )
     {
-#ifdef wxLongLong_t
+#if defined(wxLongLong_t) && wxUSE_LONLONG
         wxLongLong ll(value);
         wxString s = ll.ToString();
 #else
@@ -302,7 +302,7 @@ bool wxAnyValueTypeImplUint::ConvertValue(const wxAnyValueBuffer& src,
     wxAnyBaseUintType value = GetValue(src);
     if ( wxANY_VALUE_TYPE_CHECK_TYPE(dstType, wxString) )
     {
-#ifdef wxLongLong_t
+#if defined(wxLongLong_t) && wxUSE_LONGLONG
         wxULongLong ull(value);
         wxString s = ull.ToString();
 #else
-- 
1.9.5.msysgit.0

